src/reflect/scala/reflect/internal/tpe/TypeMaps.scala

```diff
@@ -236,10 +238,18 @@ private[internal] trait TypeMaps {
 
     /** Map this function over given list of symbols */
     def mapOver(origSyms: List[Symbol]): List[Symbol] = {
+      val firstChange = firstChangedSymbol(origSyms)
       // fast path in case nothing changes due to map
-      if (noChangeToSymbols(origSyms)) origSyms
-      // map is not the identity --> do cloning properly
-      else cloneSymbolsAndModify(origSyms, TypeMap.this)
+      if (firstChange < 0) origSyms
+      else {
+        // map is not the identity --> do cloning properly
+        val cloned = cloneSymbols(origSyms)
```

Comment:
This is correct in that all symbols do need to be cloned, as `cloneSymbols` will substituted the clones correctly into each others infos. e.g. in:

```scala
class C[A <: B, B]
```

After cloning the tparams, we need:

```scala
[A' <: B', B']
```

Reply:
Yep. Found that out the hard way, but it makes sense.